stock_product_pack/tests/test_stock_product_pack.py

```diff
@@ -89,3 +106,39 @@ def test_compute_quantities_dict(self):
         self.product_obj.invalidate_cache()
         self.assertEqual(self.pack_dc.virtual_available, 5)
         self.assertEqual(self.pack_dc.qty_available, 5)
+
+    def test_pack_with_dont_move_the_parent(self):
+        """ Run a procurement for prod pack products when there are only 5 in stock then
+        check that MTO is applied on the moves when the rule is set to 'mts_else_mto'
```

Comment:
Why you don't simply create a sale order with the detailed pack and confirm it? It's as simple as using the form class:

```python
self.pack_dc.dont_create_move = True
sale_form = Form(self.env["sale.order"])
sale_form.partner_id = self.env["res.partner"].create({"name": "Mr. Odoo"})
with sale_form.order_line.new() as line:
    line.product_id = self.pack_dc
sale = sale_form.save()
sale.action_confirm()
# And now do the assertions
self.assertFalse(self.pack_dc in sale.picking_ids.move_lines.product_id)
```

You can create another test with the oposite case as well

Reply:
Yes i realised that dependency of sale is not available here for this module. I've try some ideas, but they didn't work.
Any suggest to do it using stock orderpoint?

Reply:
The simplest way I come up with would be to create this rule and then run the scheduler (after configuring the pack product as storable and setting up an orderpoint):

![image](https://user-images.githubusercontent.com/5040182/126463094-49310e9f-5001-49e6-a7fc-7bebf8e3ab69.png)